blackduck/Client.py

```diff
@@ -200,37 +197,45 @@ def get_metadata(self, name, parent=None, **kwargs):
         kwargs['params'] = {'limit': 1}
         return self.get_resource(name, parent, items=False, **kwargs)
 
-    def checked_request(self, method, url, name='', **kwargs):
-        """Request from endpoint and return json result
+    def get_json(self, url, **kwargs):
+        """Streamline get request to url endpoint and return json result
 
         Args:
-            method (str): http request type e.g. 'get', 'post', etc.
             url (str): of endpoint
-            name (str, optional): for informational purposes in case of error. Defaults to ''.
             kwargs: passed to session.request
 
         Returns:
             json/dict: requested object
+
+        Raises:
+            requests.exceptions.HTTPError: from response.raise_for_status()
+            json.JSONDecodeError: if response.text is not json
         """
-        response = self.session.request(method, url, **kwargs)
+        r = self.session.get(url, **kwargs)
 
-        if response.status_code != 200:
-            self.http_exception_handler(response, name)
+        if r.status_code != 200:
+            # print out a more descriptive error message before raising an exception
+            self.log_http_error(r)
```

Comment:
Note when GET'ing data it pretty much either works or it does not. The user needs a descriptive error message so that they can adjust their code as necessary to make it succeed but apart from that, I do not see a compelling need for different exception types to allow error handling branching.

Reply:
I am open to re-introduce the http_exception_handler if we in the future add more functionality to it but as it stands right now it contains a whole Exception hierarchy that isn't even used.